feat: upgrade to USWDS 3.11.0

mdmower-csnw opened the pull request:
_This pull request builds off of the work in #3265. When that pull request is merged, this will be rebased and moved out of draft status._

# Summary

Apply updates related to features and bug fixes released by USWDS in [3.11.0](https://github.com/uswds/uswds/releases/tag/v3.11.0).

- Update default error message for file input and apply to `aria-label` on error

## Related Issues or PRs

(none)

## How To Test

Confirm things work the same as described in USWDS release notes: [3.11.0](https://github.com/uswds/uswds/releases/tag/v3.11.0).

## Happo differences

TODO

brandonlenz commented:
I saw the patch notes changes to input group classes handling, and tested `<InputGroup className="usa-input-group--sm">` on one of the InputPrefix examples. Works as expected.

mdmower-csnw commented:
@brandonlenz - Just an update on the status of this PR: I added the remaining TODO items to the top of the PR description. I added explanations for most of the Happo differences to the bottom of the description, but I don't have an iPhone/Mac to investigate why the slight margin appeared on iOS Safari for a few page templates. If I were to guess, it's probably related to either the footer spacing changes in https://github.com/uswds/uswds/pull/6237 or the normalization changes in https://github.com/uswds/uswds/pull/5555.

Aside from understanding that change, the one item that requires actual work is "In Page Navigation" stories. Several stories exist in USWDS that don't exist here (including nested headers nav, where I should revert my changes and try to match USWDS): https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/mh/port-banner/?path=/story/components-in-page-navigation--default

mdmower-csnw commented on src/components/InPageNavigation/InPageNavigation.tsx:
```diff
@@ -1,14 +1,68 @@
+        'Only class and id selectors are supported by InPageNavigation contentSelector'
+      )
+      return headings
+    }
```
This manual selector matching isn't something I'm proud of, especially since it's limited to id and class selectors. If in-page nav links should be built using React nodes (`<Link>{heading}</Link>` below), then trying to match like this pre-render seems to be the only option (would love to be proven incorrect here!).

I can kind of defend the "Only class and id selectors" support by a literal reading of [USWDS in-page nav documentation](https://designsystem.digital.gov/components/in-page-navigation/#using-the-in-page-navigation-component-2):
> **Custom main content region selector.** The main content selector property (`data-main-content-selector`) allows you to define which element the component will look in to pull its list of headings. Set the property’s value to the `class` or `id` of your page’s main content region (for example, `data-main-content-selector=".main-content"` or `data-main-content-selector="#main-content"`). By default, the component will pull headings from the `<main>` element.

An alternative approach to this problem would be to build the in-page nav after render, using the DOM. This is decidedly not very React-y, though. https://github.com/mdmower-csnw/react-uswds/commit/1d32da2f24cc07815e0d97a45aa2bbf7ca695952

brandonlenz reviewed (approved):
Excellent. Thank you!
